lib/improver/tests/utilities/test_cube_extraction.py

```diff
@@ -79,21 +128,26 @@ def setUp(self):
 
     def test_basic_no_units(self):
         """ Test simple key-value splitting with no units """
-        cdict, udict = parse_constraint_list(self.constraints)
+        result, udict = parse_constraint_list(self.constraints)
+        self.assertIsInstance(result, iris.Constraint)
+        self.assertEqual(
+            result._coord_values.keys(), ["threshold", "percentile"])
+        cdict = result._coord_values
         self.assertEqual(cdict["percentile"], 10)
         self.assertEqual(cdict["threshold"], 0.1)
         self.assertFalse(udict)
 
     def test_whitespace(self):
         """ Test constraint parsing with padding whitespace """
         constraints = ["percentile = 10", "threshold = 0.1"]
-        cdict, _ = parse_constraint_list(constraints)
+        result, _ = parse_constraint_list(constraints)
```

Comment:
You're right that fundamentally this is a result of the way that argparser uses spaces to split the arguments. For the purposes of this CLI, the following would work to handle white space based on the fact that these spaces are associated with punctuation marks:
```python
    parsed_constraints = []
    punctuation = ["=", ",", ":"]
    for index, constraint in enumerate(constraints):
        if constraint in punctuation:
            constraint = "".join(constraints[index-1:index+2])
        elif any(constraint.startswith(item) for item in punctuation):
            constraint = "".join(constraints[index-1:index+1])
        elif any(constraint.endswith(item) for item in punctuation):
            constraint = "".join(constraints[index:index+2])
        elif any(item in constraint for item in punctuation):
            pass
        else:
            continue
        parsed_constraints.append(constraint)
```
I'd be happy to add this, along with unit tests, if you think is sensible.

Reply:
Let's see if the second reviewer thinks it is worthwhile. If not we will leave it for now.

Reply:
Since Gavin has a solution, we should probably implement it now rather than accumulating technical debt to fix it later.

Reply:
Regarding the' raising of warnings when selected ranges go beyond the limits of the cube' issue, I would have thought that this would already be a feature of Iris. I am rather surprised that you can ask for limits outside the data ranges in the cube without at least being warned about it. Probably is worth raising this with the Iris developers.

Reply:
I'm happy to try to add the white space handler, although after some testing of the suggested code above, it is a bit fragile and needs a bit for work and extra functionality. If that is the only outstanding work required for this PR, then I think I'd recommend that we merge this PR, and I could work on adding white space handling in a separate PR, if it's required.

Reply:
@bayliffe's agreed that we can leave the white space handling for now, as it's not ready for inclusion within this pull request. 

For reference, the current function I have is shown below, although this doesn't currently handle situations where the user has specified multiple constraints with unnecessary white space, for example, "name = air_temperature projection_y_coordinate = [0 : 2000]" with the function below returning a confused list of constraints:
```python
def white_space_handler(constraints):
    """
    Handle arguments that are specified on the command line with unnecessary
    white space. This function parses for punctuation marks that are
    indicative that the user has provided redundant white space, and joins
    together command line arguments that should be treated as a single
    constraint.

    White space handling that is supported is of the form:
        name = altitude <- Extra white space surrounding "=" sign.
        projection_y_coordinate=[-398000 : -160000] <- Extra white space
        surrounding ":".
        projection_x_coordinate=[0, 2000] <- Extra white space surrounding ",".

    Args:
        constraints (list):
            List of string constraints with keys and values split by "=":
            e.g: ["kw1=val1", "kw2 = val2", "kw3=val3"].

    Returns:
        parsed_constraints (list):
            List of string constraints with keys and values split by "=":
            e.g: ["kw1=val1", "kw2 = val2", "kw3=val3"].
    """

    def find_partial_match(constraints, constr):
        matches = []
        for x in constraints:
            if constr in x and constr != x:
                matches.append(True)
            else:
                matches.append(False)
        return matches

    parsed_constraints = []
    punctuation = ["=", ",", ":"]
    for index, constraint in enumerate(constraints):
        if constraint in punctuation:
            constraint = "".join(constraints[index-1:index+2])
        elif any(constraint.startswith(item) for item in punctuation):
            constraint = "".join(constraints[index-1:index+1])
        elif any(constraint.endswith(item) for item in punctuation):
            constraint = "".join(constraints[index:index+2])
        elif any(item in constraint for item in punctuation):
            pass
        else:
            continue
        parsed_constraints.append(constraint)

    for constr in parsed_constraints:
        if any(find_partial_match(parsed_constraints, constr)):
            parsed_constraints.remove(constr)
    return parsed_constraints
```